src/SFML/Window/DRM/DRMContext.cpp

```diff
@@ -130,6 +128,308 @@ namespace
         initialized = false;
     }
 
+    void drmFbDestroyCallback(gbm_bo* bo, void* data)
+    {
+        int drmFd = gbm_device_get_fd(gbm_bo_get_device(bo));
+        sf::priv::DrmFb* fb = reinterpret_cast<sf::priv::DrmFb*>(data);
+
+        if (fb->fbId)
+            drmModeRmFB(drmFd, fb->fbId);
+
+        std::free(fb);
+    }
+
+    sf::priv::DrmFb* drmFbGetFromBo(gbm_bo& bo)
+    {
+        int drmFd = gbm_device_get_fd(gbm_bo_get_device(&bo));
+        sf::priv::DrmFb* fb = reinterpret_cast<sf::priv::DrmFb*>(gbm_bo_get_user_data(&bo));
+        if (fb)
+            return fb;
+
+        fb = reinterpret_cast<sf::priv::DrmFb*>(std::calloc(1, sizeof(*fb)));
```

Comment:
I'd prefer to replace this with `operator new` but I'm not sure if `new + memset` is better than `calloc + reinterpret_cast`. I seldom write code so low level so hopefully someone can help find a better way to express with within the limits of C++03.

Reply:
This should not be `reinterpret_cast`, anyway.
[`std::calloc()`](https://en.cppreference.com/w/cpp/memory/c/calloc) returns `void*`, which is convertible to other pointer types via `static_cast`.

But couldn't we just do this?
```cpp
fb = new DrmFb();
```
That type is defined as a POD:
https://github.com/SFML/SFML/blob/22033670f8800f077c2f4cd6419b4405feec1240/src/SFML/Window/DRM/DRMContext.hpp#L59-L63

All members (also PODs) will be zero-initialized.

Reply:
Are you sure they're zero-initialized? My impression was that they're left uninitialized.

Reply:
The subtle difference between:
```cpp
fb = new DrmFb; // uninit
fb = new DrmFb(); // zero
```
😉 

It also works for arrays:
```cpp
ptr = new int[5]; // uninit
ptr = new int[5](); // zero
```

Reply:
https://godbolt.org/z/jYn3Kq398

That's cool, I didn't know that! In that case this is an easy fixup then.